File: smarr_are_we_fast_yet/benchmarks/JavaScript/nbody.run.js
let bm = require('./nbody.js');

let benchmark = bm.newInstance();
let n = parseInt(process.argv[2], 10);

const system = new bm.NBodySystem();
for (let i = 0; i < n; i += 1) {
    system.advance(0.01);
}
console.log(system.energy().toFixed(6));

File: smarr_are_we_fast_yet/benchmarks/Lua/nbody.run.lua
package.path = './smarr_are_we_fast_yet/benchmarks/Lua/?.lua;' .. package.path
local benchmark = require('nbody')
local inner_iterations = arg[1] + 0

local system = benchmark.NBodySystem.new()
for _ = 1, inner_iterations do
    system:advance(0.01)
end
print (string.format("%f", system:energy()))